permissions/perUserMfaState/grantPermission.ps1

```diff
@@ -0,0 +1,322 @@
+    # Microsoft docs: https://learn.microsoft.com/nl-nl/graph/api/PerUserMfaState-get?view=graph-rest-1.0&tabs=http
+    $actionMessage = "querying perUserMfaState for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json)"
+
+    $baseUri = "https://graph.microsoft.com/"
+    $getCurrentperUserMfaStateSplatParams = @{
+        Uri         = "$($baseUri)/beta/users/$($actionContext.References.Account)/authentication/requirements"
+        Headers     = $headers
+        Method      = "GET"
+        Verbose     = $false
+        ErrorAction = "Stop"
+    }
+
+    $currentPerUserMfaState = $null
+    $currentPerUserMfaState = (Invoke-RestMethod @getCurrentperUserMfaStateSplatParams)
+
+    $currentPerUserMfaState = $currentPerUserMfaState.perUserMfaState
+
+    Write-Verbose "Queried perUserMfaState for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Result: $($currentPerUserMfaState | ConvertTo-Json)"
+    #endregion Get current PerUserMfaState
+
+    #region Calulate action
+    $actionMessage = "calculating action"
+    
+    # Remove spaces
+    
+    if ($currentPerUserMfaState -eq 'disabled') {
+        $actionPerUserMfaState = "Update"
+    }
+    else {
+        $actionPerUserMfaState = "NoChanges"
+    }
+    
+    #endregion Calulate action
+
+    #region Process
+    switch ($actionPerUserMfaState) {
+        "Update" {
+            #region Update PerUserMfaState
+            # Microsoft docs: https://learn.microsoft.com/nl-nl/graph/api/PerUserMfaState-update?view=graph-rest-1.0&tabs=http
+            $actionMessage = "setting perUserMfaState to [$($state)] for account"
+            $baseUri = "https://graph.microsoft.com/"
+
+            $updatePerUserMfaStateBody = @{
+                "perUserMfaState" = $state
+            }
+
+            $updatePerUserMfaStateSplatParams = @{
+                Uri         = "$($baseUri)/beta/users/$($actionContext.References.Account)/authentication/requirements"
+                Method      = "PATCH"
+                Body        = ($updatePerUserMfaStateBody | ConvertTo-Json -Depth 10)
+                Verbose     = $false
+                ErrorAction = "Stop"
+            }
+
+            Write-Verbose "SplatParams: $($updatePerUserMfaStateSplatParams | ConvertTo-Json)"
+
+            if (-Not($actionContext.DryRun -eq $true)) {
+                # Add Headers after printing splat
+                $updatePerUserMfaStateSplatParams['Headers'] = $headers
+
+                $updatedPerUserMfaState = Invoke-RestMethod @updatePerUserMfaStateSplatParams
+
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "$state perUserMfaState [$($actionContext.References.Permission.Name)] for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Old value: [$($currentPerUserMfaState)]. New value: [$($state)]."
```

Comment:
new state heb je al voorin de message staan. is het nodig om dit 2x in het audit bericht te plaatsen?

Reply:
aangepast